Sources/FoundationEssentials/String/String+Internals.swift

```diff
@@ -138,19 +138,20 @@ extension String {
             return try block(buffer.baseAddress!)
         }
         #else
-        var iter = self.utf8.makeIterator()
 #if os(Windows)
+        var iter = self.utf8.makeIterator()
         let bLeadingSlash = if iter.next() == ._slash, iter.next()?.isLetter ?? false, iter.next() == ._colon { true } else { false }
-#else
-        let bLeadingSlash = false
-#endif
-
         // Strip the leading `/` on a RFC8089 path (`/[drive-letter]:/...` ).  A
         // leading slash indicates a rooted path on the drive for the current
         // working directory.
-        return try Substring(self.utf8.dropFirst(bLeadingSlash ? 1 : 0)).withCString {
+        return try Substring(self.utf8.dropFirst(bLeadingSlash ? 1 : 0)).replacing("/", with: "\\").withCString {
```

Comment:
There's an internal `StringProtocol` extension that's much faster if you're replacing single ascii characters: `internal func replacing(_ a: UInt8, with b: UInt8)`